src/components/SingleArticle.jsx

```diff
@@ -4,14 +4,44 @@ import { GetSingleArticle } from "../apis/apis";
 import { useParams, Link } from "react-router-dom";
 import Comments from "./Comments/Comments";
 
+import ThumbUpIcon from "@mui/icons-material/ThumbUp";
+import ThumbDownIcon from "@mui/icons-material/ThumbDown";
+import { voteOnArticles } from "../apis/apis";
+
 function SingleArticle() {
   const { id } = useParams();
   const [article, setArticle] = useState([]);
   const [loading, setisLoading] = useState(true);
+  const [prevCounts, setPrevCounts] = useState(0);
+
+  const [newVote, setNewVote] = useState({ incVotes: 1 });
+
+  const handleOnClick = () => {
+    setNewVote({ incVotes: 1 });
+    voteOnArticles(id, newVote)
+      .then((res) => {
+        setPrevCounts(res + 1);
+      })
+      .catch((err) => {
+        console.log(err, "<-- err");
```

Comment:
We should be handling the vote change optimistically which means setting the new vote amount before we have had a successful api request and then reverting it back in the catch block if it is unsuccessful. You can see an example of optimistic rendering in the notes here - https://notes.northcoders.com/courses/js-front-end/ui-behaviours-and-optimistic-rendering#optimistic-rendering